src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+                    }
+                }
+
+                return aborted;
+            }
+
+            //bool WaitForInputSessionClose(TimeSpan timeout, out bool wasAborted)
+            //{
+            //    Message message;
+            //    TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            //    wasAborted = false;
+            //    try
+            //    {
+            //        if (this.TryReceive(timeoutHelper.RemainingTime(), out message))
+            //        {
+            //            if (message != null)
+            //            {
+            //                using (message)
+            //                {
+            //                    ProtocolException error = ProtocolException.ReceiveShutdownReturnedNonNull(message);
+            //                    throw TraceUtility.ThrowHelperWarning(error, message);
+            //                }
+            //            }
+            //            return this.inputSessionClosedHandle.Wait(timeoutHelper.RemainingTime(), false);
+            //        }
+            //    }
+            //    catch (CommunicationObjectAbortedException)
+            //    {
+            //        if (this.State != CommunicationState.Closed)
+            //        {
+            //            throw;
+            //        }
+            //        wasAborted = true;
+            //    }
+            //    return false;
+            //}
+
+            protected override void OnCloseResponseMessageReceived(RequestContext requestContext, Message message, SecurityProtocolCorrelationState correlationState, TimeSpan timeout)
+            {
+                // we dont expect a close-response for non-duplex security session
+                message.Close();
+                requestContext.Abort();
+                this.Fault(new ProtocolException(SR.Format(SR.UnexpectedSecuritySessionCloseResponse)));
+            }
+
+            private void Fault(ProtocolException protocolException)
+            {
+                this.AbortCore();
+                //    this.inputSessionClosedHandle.Fault(this);
+                base.OnFaulted(protocolException);
+
+
+            }
+
+            protected override void OnCloseMessageReceived(RequestContext requestContext, Message message, SecurityProtocolCorrelationState correlationState, CancellationToken token)
+            {
+                if (this.State == CommunicationState.Created)
+                {
+                    Fx.Assert("ServerSecuritySimplexSessionChannel.OnCloseMessageReceived (this.State == Created)");
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ServerReceivedCloseMessageStateIsCreated, this.GetType().ToString())));
+                }
+
+                if (SendCloseResponseOnCloseReceivedIfRequired(requestContext, message, correlationState, token))
+                {
+                  //  this.inputSessionClosedHandle.Set();
```

Comment:
This does look like we should have something wired up to handle closing.

Reply:
let's talk same as https://github.com/CoreWCF/CoreWCF/pull/172#discussion_r525612609